### DISCUSSION TITLE
Rewriting PHPDocs based on the correct types

### DISCUSSION THREAD

**Discussion by mabumusa1** [symfony/symfony#50693] [Q&A]
Hello all, 

This might be a long shot, but maybe someone can help or have ideas.

In [Mautic](https://github.com/mautic/mautic), we are doing a lot of refactoring at the moment to upgrade the code base to be `PHP 8.x` one of the issues I am facing personally is that we have lots of classes with the wrong PHPDocs which makes it impossible for tools like `rector` to work correct since the types are either wrong or uses `mixed` which is like there is no type at all.

I have done some research but I was not lucky finding any tool that would analyze the code and then regenerate `PHPDocs` using the write types.

If someone has suggestions, it would be super helpful.

Thank you

**Comment by ghost** [comment]
This discussion is not Symfony related.

P.S. It doesn't make any sense to analyze code to add PHPDoc blocks that would then be used to typehint a codebase.
Rector has "rectors" that can add typehints based on code analysis.
